src/tests/ftest/container/open.py

```diff
@@ -26,107 +26,141 @@
+            "Case 3: Test Bug!"
+        ]
+        result_messages = [messages_case_1, messages_case_2, messages_case_3]
+
+        # Create the pool and connect. Then create the container from the pool
+        # Add the pool and the container created into a list
+        container_uuids = []
+        for i in range(2):
+            self.pool.append(TestPool(self.context, self.log))
+            self.pool[-1].get_params(self)
+            self.pool[-1].create()
+            self.pool[-1].connect(1)
+            self.container.append(TestContainer(self.pool[-1]))
+            self.container[-1].get_params(self)
+            self.container[-1].create()
+            container_uuids.append(uuid.UUID(self.container[-1].uuid))
+
+        # Decide which pool handle and container UUID to use. The PASS/FAIL
+        # number corresponds to the index for self.pool and container_uuids
+        pool_handle_index = poh_state_num
+        container_uuid_index = uuid_state_num
+
+        # Case 1
         try:
-            # Create and connect to two pools
-            for index in range(2):
-                self.pool.append(DaosPool(self.context))
-                self.pool[index].create(
-                    mode, uid[index], gid[index], size, name, None)
-                self.pool[index].connect(1 << 1)
-
-            # defines pool handle for container open
-            if pohlist[0] == 'pool1':
-                poh = self.pool[0].handle
-            else:
-                poh = self.pool[1].handle
-
-            # Create a container in the first pool
-            self.container.append(DaosContainer(self.context))
-            self.container[0].create(self.pool[0].handle)
-
-            # defines test UUID for container open
-            if uuidlist[0] == 'pool1':
-                struuid = self.container[0].get_uuid_str()
-                container_uuid = uuid.UUID(struuid)
-            else:
-                container_uuid = uuid.uuid4()   # random uuid
-
-            # Try to open the first container
-            # open should be ok only if poh = pool1.handle &&
-            #                           containerUUID = container1.uuid
-            self.container[0].open(poh, container_uuid)
-
-            # wait a few seconds and then destroy containers
-            time.sleep(5)
-            self.destroy_containers(self.container)
-
-            # cleanup the pools
-            self.destroy_pools(self.pool)
-
-            if expected_result in ['FAIL']:
-                self.fail("Test was expected to fail but it passed.\n")
-
-        except DaosApiError as excep:
-            print(excep)
+            self.container[0].open(self.pool[pool_handle_index].pool
+                                   .handle.value,
+                                   container_uuids[container_uuid_index])
+            self.assertEqual(expected_result, RESULT_PASS, 
+                             result_messages[test_case][0])
+        except TestFail as result:
```

Comment:
(pylint-redefine-in-handler) Redefining name 'result' from outer scope (line 86) in exception handler